torch/_dynamo/variables/ctx_manager.py

```diff
@@ -172,9 +172,9 @@ def create(tx, target_values, **kwargs):
         return var
 
     def enter(self, tx):
-        install_guard(self._guards_singleton)
         batch_size, randomness = self.target_values
         vmap_level = torch._C._functorch._vmap_increment_nesting(batch_size, randomness)
+        install_guard(self._guards_singleton)
```

Comment:
Why install the guard after entering? I would expect to install the guard before